### PR TITLE
Fix doc blocks typing for TwiML

### PR DESCRIPTION
**Pull request by erickskrauch**
I'm not sure about this, I saw you use some code generation to maintain the library, but this is a concept of the typing that should be made to allow IDE autocompletion and static analysis work.

### CODE REVIEW

**Comment by peter279k** [comment]
It looks like some files have conflicts. They need to be fixed.

**Comment by erickskrauch** [comment]
@peter279k should be good now. I also fixed the types for the parameters. But you still need to adjust the generator to all this automatically.

**Comment by dinamic** [comment]
@erickskrauch thanks for the improvements! Just noticed the annoyingly mistaken phpdocs in my PHPStorm and was about to submit a PR and noticed yours.

You might want to resolve those conflicts, once again.

**Comment by erickskrauch** [comment]
@dinamic done.

**Comment by dinamic** [comment]
Ping @peter279k @ekarson @cjcodes 

Guys, could we have this one merged? Is really annoying to have the IDE to warn your about non-existing methods while they are actually there and nothing's wrong with the usage.

screenshot 2018-10-23 at 17 32 26

File: Twilio/TwiML/TwiML.php [hunk 1]
```diff
@@ -71,6 +71,7 @@ public function setAttribute($key, $value) {
 	 * @param string $name XML element name
 	 * @param string $value XML value
 	 * @param array $attributes XML attributes
+     * @return TwiML
```
Comment (dinamic) [inline Twilio/TwiML/TwiML.php]: This one seem wrong. I think the return type is `GenericNode`.

File: Twilio/TwiML/Voice/Dial.php [hunk 2]
```diff
@@ -24,202 +24,200 @@ public function __construct($number, $attributes = array()) {
...
         return $this->setAttribute('recordingStatusCallbackEvent', $recordingStatusCallbackEvent);
     }
 
     /**
      * Add AnswerOnBridge attribute.
-     * 
-     * @param boolean $answerOnBridge Preserve the ringing behavior of the inbound
+     *
+     * @param bool $answerOnBridge Preserve the ringing behavior of the inbound
      *                                call until the Dialed call picks up
```
Comment (dinamic) [inline Twilio/TwiML/Voice/Dial.php]: You may want to line this one up.

**Comment by cjcodes** [comment]
@erickskrauch and @dinamic, thanks for the work on this one!

As you noted in the beginning of this thread, most of these files are generated automatically by an internal tool called Yoyodyne. These files have the [yoyodyne header](https://github.com/twilio/twilio-php/blob/master/CONTRIBUTING.md#-found-an-issue).

I've escalated this internally and linked to this PR as the reference implementation.

One question: PHP's PSR recommends using either `return static` or `return $this`, both of which seem to be well-supported in IDEs. I'm partial to being less redundant, so any opposition to using `return $this` instead?

**Comment by dinamic** [comment]
Thanks for the quick reaction, @cjcodes! 

Not sure I understand the use case for `return static;`. It's not part of the supported syntax currently.

```
~$ php -a
Interactive shell

php > class X { public function xx() { return static; } };
PHP Parse error:  syntax error, unexpected ';', expecting :: (T_PAAMAYIM_NEKUDOTAYIM) in php shell code on line 1

Parse error: syntax error, unexpected ';', expecting :: (T_PAAMAYIM_NEKUDOTAYIM) in php shell code on line 1
```

If the question is regarding what to use inside phpdoc `@return` statement, I feel like `@return $this` is well supported by most editors and IDEs.

Furthermore, I feel like phpdoc is somewhat getting obsolete as strict types make their appearance in php. With PHP 5.6 and PHP 7.0 approaching [end of life in two months](http://php.net/supported-versions.php), maybe strict typing could replace phpdoc if a newer version gets tagged. We should just make sure the variable names are as descriptive as possible.

**Comment by cjcodes** [comment]
@dinamic Sorry, I meant in the docblock. In other words, replacing this from the PR:

```
/**
 * ...
 * @return static $this.
 */
```

With this:

```
/**
 * ...
 * @return $this
 */
```

**Comment by dinamic** [comment]
@cjcodes I think using `@return $this` is well supported and should work in most if not all editors and IDEs. Thanks for catching up on this. 👍

**Comment by erickskrauch** [comment]
@dinamic I prefer to use `static` type because PHPStorm not always correctly recognize `$this`, but perfectly works with `self` and `static`.

**Comment by dinamic** [comment]
@erickskrauch thanks for the info. I wasn't aware of any issues with `$this`.

`static` should be fine then.

**Comment by childish-sambino** [comment]
Working on making changes in the upstream code generator. Splitting up into multiple PRs here and linking.

**Pull request merged**